cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -615,36 +609,48 @@ set(Boost_ADDITIONAL_VERSIONS
     "1.60.0"
     "1.60")
 
-if(BOOST_SOURCE STREQUAL "AUTO")
-  find_package(BoostAlt ${ARROW_BOOST_REQUIRED_VERSION})
-  if(NOT BoostAlt_FOUND)
-    build_boost()
-  endif()
-elseif(BOOST_SOURCE STREQUAL "BUNDLED")
-  build_boost()
-elseif(BOOST_SOURCE STREQUAL "SYSTEM")
-  find_package(BoostAlt ${ARROW_BOOST_REQUIRED_VERSION} REQUIRED)
-endif()
-
-if(TARGET Boost::system)
-  set(BOOST_SYSTEM_LIBRARY Boost::system)
-  set(BOOST_FILESYSTEM_LIBRARY Boost::filesystem)
-  set(BOOST_REGEX_LIBRARY Boost::regex)
-elseif(BoostAlt_FOUND)
-  set(BOOST_SYSTEM_LIBRARY ${Boost_SYSTEM_LIBRARY})
-  set(BOOST_FILESYSTEM_LIBRARY ${Boost_FILESYSTEM_LIBRARY})
-  set(BOOST_REGEX_LIBRARY ${Boost_REGEX_LIBRARY})
+if(ARROW_BUILD_INTEGRATION
+   OR ARROW_BUILD_TESTS
+   OR ARROW_HDFS
+   OR ARROW_GANDIVA
+   OR ARROW_PARQUET)
+  set(ARROW_BOOST_REQUIRED TRUE)
```

Comment:
Do all of these require the same boost libraries? Gandiva requires boost::filesystem?

Reply:
This flag just determines whether or not we create boost_ep (or use the system / toolchain Boost if it's available), so it's an all-or-nothing thing